### PR TITLE
Move production description from Tooltip to Buildable.

### LINKED ISSUE #12045: Move production description from Tooltip to Buildable
As pointed out in #11984.

### PR DESCRIPTION
**Pull request by tysonliddell**
Fixes #12045.

### CODE REVIEW

File: mods/ts/rules/civilian-structures.yaml [hunk 1]
```diff
@@ -1261,7 +1263,6 @@ GALITE:
 		Range: 2c0
 	Power:
 		Amount: 0
-# TODO: should use terrain lighting instead, depends on https://github.com/OpenRA/OpenRA/issues/3605
```
Comment (reaperrr) [inline mods/ts/rules/civilian-structures.yaml]: This probably slipped in after running the upgrade rules, please revert this removal.

File: OpenRA.Mods.Common/Traits/Buildable.cs [hunk 2]
```diff
@@ -45,6 +45,9 @@ public class BuildableInfo : TraitInfo<Buildable>
 
 		// TODO: UI fluff; doesn't belong here
 		public readonly int BuildPaletteOrder = 9999;
+
+		[Desc("Description moved from Tooltip.")]
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Buildable.cs]: Please restore the [Translate], you can simply put it on the line above.

File: OpenRA.Mods.Common/Traits/Buildable.cs [hunk 3]
```diff
@@ -46,7 +46,7 @@ public class BuildableInfo : TraitInfo<Buildable>
 		// TODO: UI fluff; doesn't belong here
 		public readonly int BuildPaletteOrder = 9999;
 
-		[Desc("Description moved from Tooltip.")]
+		[Desc("Text shown in the production tooltip.")]
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Buildable.cs]: I think you should just fixup this commit into the first.

**Comment by reaperrr** [comment]
I haven't tested this yet, but code and yaml looks good to me once the minor things I pointed out are dealt with.

**Comment by tysonliddell** [comment]
Issues pointed out by reaperrr have been fixed up.

**Comment by reaperrr** [comment]
One remaining issue and one "optional, but would be nice" thing:
- you forgot or accidentally discarded the changes to TS' `gdi-structures.yaml`.
- while it lacks one on bleed, it would be nice if you could add a simple `Produces structures.` desc to the D2K construction yard while you're at it.

:+1: after that.

**Comment by tysonliddell** [comment]
Fixed up both additional points by reaperrr

**Review by reaperrr — APPROVED** [review]

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 4]
```diff
@@ -386,6 +386,25 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Move production description from Tooltip to Buildable
+                if (engineVersion < 20161001)
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: Can you please bump the date here while you're at it?

**Review by pchote — CHANGES_REQUESTED** [review]
Sorry for the delay in looking at this, it looks good to merge but needs a rebase.  Let us know on IRC if you need help, or if you run into problems one of us can do the rebase and then push directly to your branch.

**Review by pchote — APPROVED** [review]

**Comment by tysonliddell** [comment]
Rebased and date in UpgradeRules.cs updated.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/bb10efb20aed29091ff036b9d9d3dbdb6edb8842

**Pull request merged**